packages/medusa/src/services/shipping-option.ts

```diff
@@ -0,0 +1,717 @@
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          "is_return cannot be changed after creation"
+        )
+      }
+
+      if (typeof update.requirements !== "undefined") {
+        const acc: ShippingOptionRequirement[] = []
+        for (const r of update.requirements) {
+          const validated = await this.validateRequirement_(r, optionId)
+
+          if (acc.find((raw) => raw.type === validated.type)) {
+            throw new MedusaError(
+              MedusaError.Types.INVALID_DATA,
+              "Only one requirement of each type is allowed"
+            )
+          }
+
+          if (
+            acc.find(
+              (raw) =>
+                (raw.type === "max_subtotal" &&
+                  validated.amount > raw.amount) ||
+                (raw.type === "min_subtotal" && validated.amount < raw.amount)
+            )
+          ) {
+            throw new MedusaError(
+              MedusaError.Types.INVALID_DATA,
+              "Max. subtotal must be greater than Min. subtotal"
+            )
+          }
+
+          acc.push(validated)
+        }
+
+        if (option.requirements) {
+          const accReqs = acc.map((a) => a.id)
+          const toRemove = option.requirements.filter(
+            (r) => !accReqs.includes(r.id)
+          )
+          await Promise.all(
+            toRemove.map(async (req) => {
+              await this.removeRequirement(req.id)
+            })
+          )
+        }
+
+        option.requirements = acc
+      }
+
+      if (typeof update.price_type !== "undefined") {
+        option.price_type = await this.validatePriceType_(
+          update.price_type,
+          option
+        )
+        if (update.price_type === "calculated") {
+          option.amount = null
+        }
+      }
+
+      if (
+        typeof update.amount !== "undefined" &&
+        option.price_type !== "calculated"
+      ) {
+        option.amount = update.amount
+      }
+
+      if (typeof update.name !== "undefined") {
+        option.name = update.name
+      }
+
+      if (typeof update.admin_only !== "undefined") {
+        option.admin_only = update.admin_only
+      }
+
+      const optionRepo = manager.getCustomRepository(this.optionRepository_)
+      return await optionRepo.save(option)
+    })
+  }
+
+  /**
+   * Deletes a profile with a given profile id.
+   * @param {string} optionId - the id of the profile to delete. Must be
+   *   castable as an ObjectId
+   * @return {Promise} the result of the delete operation.
+   */
+  async delete(optionId: string): Promise<ShippingOption | void> {
+    try {
+      const option = await this.retrieve(optionId)
+
+      const optionRepo = this.manager_.getCustomRepository(
+        this.optionRepository_
+      )
+
+      return optionRepo.softRemove(option)
+    } catch (error) {
+      // Delete is idempotent, but we return a promise to allow then-chaining
+      return Promise.resolve()
```

Comment:
very nit(non-blocking)
```suggestion
      return
```